SU2_CFD/src/output/CMultizoneOutput.cpp

```diff
@@ -125,6 +148,54 @@ void CMultizoneOutput::SetMultizoneHistoryOutputFields(const COutput* const* out
         AddHistoryOutput(name, header, field.screenFormat, group, "", field.fieldType );
       }
     }
+
+    /*--- Prepare Marker lists that are passed to 'AddHistoryOutputPerSurface'. ---*/
+    vector<string> Marker_Analyze;
+    for (unsigned short iMarker_Analyze = 0; iMarker_Analyze < config[iZone]->GetnMarker_Analyze(); iMarker_Analyze++) {
+      Marker_Analyze.push_back(config[iZone]->GetMarker_Analyze_TagBound(iMarker_Analyze));
+    }
+
+    vector<string> Marker_Monitoring;
+    for (unsigned short iMarker_Monitoring = 0; iMarker_Monitoring < config[iZone]->GetnMarker_Monitoring(); iMarker_Monitoring++) {
+      Marker_Monitoring.push_back(config[iZone]->GetMarker_Monitoring_TagBound(iMarker_Monitoring));
+    }
+
+    /*--- Add the PerSurface outputs. ---*/
+    const auto& ZoneHistoryPerSurfaceFields = output[iZone]->GetHistoryPerSurfaceFields();
+
+    for (const auto& nameSinglezone : output[iZone]->GetHistoryOutputPerSurface_List()) {
+
+      const auto& field = ZoneHistoryPerSurfaceFields.at(nameSinglezone);
+
+      name = nameSinglezone + zoneIndex;
+
+      /*--- Remove the unnecessary Marker name from the fieldName, i.e. "Avg_Massflow(inlet)"->"Avg_Massflow". ---*/
+      /*--- Note that index zero in 'field[0]' refers to a specific Marker. Some attributes remain constant over the markers
+            like the first part of the name, the screenFormat and the fieldType. ---*/
+      string baseheader = field[0].fieldName;
+      std::string::size_type pos = baseheader.find('(');
+      if (pos != std::string::npos)
+        baseheader = baseheader.substr(0, pos);
+      else
+        SU2_MPI::Error("Cannot process PerSurface *_SURF history output: " + baseheader, CURRENT_FUNCTION);
+
+      header = baseheader + zoneIndex;
+      /*--- Attach zone-index to the group after determining which group it is. ---*/
+      group = field[0].outputGroup;
+
+      /*--- Determine whether Maker_Analyze/Monitoring has to be used. ---*/
+      vector<string> Marker;
+      if (group == "FLOW_COEFF_SURF")
+        Marker = Marker_Analyze;
```

Comment:
find a way to avoid copying a vector of strings

Reply:
... found a working `*|&` permutation that does the job in 2cbbe23